confidence/formats.py

```diff
@@ -0,0 +1,78 @@
+import json
+import typing
+from abc import abstractmethod
+from dataclasses import dataclass, replace
+from os import PathLike
+from pathlib import Path
+
+import yaml
+
+from confidence.models import unwrap
+
+
+@dataclass(frozen=True)
+class Format(typing.Protocol):
+    suffix: str = ''
```

Comment:
if it is a protocol, why would you place this default?

Reply:
This implementation is currently trying to be both. A protocol would make more sense if it was just definitions. Moved it to an ABC.